Sources/DistributedSystem/DistributedSystem.swift

```diff
@@ -960,21 +964,29 @@ public class DistributedSystem: DistributedActorSystem, @unchecked Sendable {
                 var continuation: AsyncStream<InvocationEnvelope>.Continuation?
                 let stream = AsyncStream(InvocationEnvelope.self, bufferingPolicy: .unbounded) { continuation = $0 }
                 guard let continuation else { fatalError("Internal error: continuation unexpectedly nil") }
-                self.actors[targetID] = .clientForRemoteService(channel, continuation)
-                Task { await self.streamTask(stream, actor, channel) }
-                return continuation
-            case let .serviceForRemoteClient(_, continuation):
-                return continuation
-            case let .clientForRemoteService(_, continuation):
-                return continuation
+                let queueSize = ManagedAtomic<UInt64>(0)
+                self.actors[targetID] = .clientForRemoteService(channel, continuation, queueSize)
+                Task { await self.streamTask(stream, actor, channel, queueSize) }
+                return (continuation, queueSize)
+            case let .serviceForRemoteClient(_, continuation, queueSize):
+                return (continuation, queueSize)
+            case let .clientForRemoteService(_, continuation, queueSize):
+                return (continuation, queueSize)
             default:
                 logger.error("\(channel.remoteAddressDescription): invalid actor state \(actorInfo)")
                 return nil
             }
         }
 
-        if let continuation {
-            continuation.yield(envelope)
+        if let res {
+            let sizeBits = (UInt64.bitWidth - 8)
+            let sizeMask = ((UInt64(1) << sizeBits) - 1)
+            let queueSize = res.queueSize.wrappingIncrementThenLoad(by: envelope.size, ordering: .releasing)
+            if (queueSize & sizeMask) > (Self.endpointQueueWarningSize << (queueSize >> sizeBits)) {
```

Comment:
This is a bit unreadable, but does this properly account for not logging continuously, but only when the target queue size is over the warning threshold?

Reply:
The warning threshold is multiplied by 2 each time it was breached.
So, warning will appear on 1, 2, 4, 8... etc megabytes for each particular endpoint.